### PR TITLE
bug 1784052: delete secrets if immutable fields don't match

### PR DESCRIPTION
**Pull request by deads2k**
secret types are immutable, so we need to delete and recreate on certain kinds of errors.

/hold

hold for proof.

### CODE REVIEW

**Comment by deads2k** [comment]
/assign @mfojtik @tnozicka 

proof with test passed https://github.com/openshift/cluster-kube-apiserver-operator/pull/697

File: pkg/operator/resource/resourceapply/core.go [hunk 1]
```diff
@@ -252,8 +252,24 @@ func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, re
 		klog.Infof("Secret %s/%s changes: %v", required.Namespace, required.Name, JSONPatchSecretNoError(existing, existingCopy))
 	}
 	actual, err := client.Secrets(required.Namespace).Update(existingCopy)
-
 	reportUpdateEvent(recorder, existingCopy, err)
+
+	if err == nil {
+		return actual, true, err
+	}
+	if !strings.Contains(err.Error(), "field is immutable") {
```
Comment (mfojtik) [inline pkg/operator/resource/resourceapply/core.go]: nit: can you move this up? `if err == nil || !string....`

**Pull request merged**